nifi-nar-bundles/nifi-deeplearning4j-bundle/nifi-deeplearning4j-processors/src/main/java/org/apache/nifi/processors/deeplearning4j/DeepLearning4JPredictor.java

```diff
@@ -0,0 +1,218 @@
+            }
+
+            MultiLayerNetwork model = getModel(context);
+
+            long startTimeMillis = System.currentTimeMillis();
+
+            String [] inputRecords = input.split(recordSeparator);
+
+            List<INDArray> features = Arrays.stream(inputRecords).map(
+                record -> {
+                    double [] parameters = Arrays.stream(record.split(fieldSeparator)).mapToDouble(
+                             field -> Double.parseDouble(field)).toArray();
+
+                    INDArray featureInput = Nd4j.create(parameters, dimensions);
+
+                    if ( getLogger().isDebugEnabled() ) {
+                        getLogger().debug("Features for record {} parameters {} dims {} featureInput {} ",
+                            new Object[] {record, parameters, dimensions, featureInput});
+                    }
+
+                    return featureInput;
+
+                }).collect(Collectors.toList());
+
+           INDArray allFeatures = Nd4j.vstack(features);
+
+           INDArray results = model.output(allFeatures);
+
+           double [][] partitionedResults = new double[inputRecords.length][];
+           for (int row = 0; row < inputRecords.length; row++) {
+                INDArray result = results.getRow(row);
+                partitionedResults[row] = Nd4j.toFlattened(result).toDoubleVector();
+           }
+
+           String jsonResult = gson.toJson(partitionedResults);
+           int [] shape = results.shape();
+           String jsonShape = gson.toJson(Arrays.copyOfRange(shape, 1, shape.length));
+
+           if ( getLogger().isDebugEnabled() ) {
+                getLogger().debug("Prediction for inputRecords {}, dims {}, results {}, result.shape {}, partitionedResults {}, jsonResult {}, shape {}, jsonShape {}",
+                        new Object[] {inputRecords, dimensions, results, Arrays.toString(results.shape()), partitionedResults, jsonResult, shape, jsonShape});
+           }
+
+           try (ByteArrayInputStream bais = new ByteArrayInputStream(jsonResult.getBytes(charset))) {
+               flowFile = session.importFrom(bais, flowFile);
+           }
+
+           session.putAttribute(flowFile, DEEPLEARNING4J_OUTPUT_SHAPE, jsonShape);
+
+           final long endTimeMillis = System.currentTimeMillis();
+
+           session.transfer(flowFile, REL_SUCCESS);
+
+           session.getProvenanceReporter().send(flowFile, makeProvenanceUrl(context),
+                    (endTimeMillis - startTimeMillis));
+        } catch (Exception exception) {
+            flowFile = populateErrorAttributes(session, flowFile, exception.getMessage());
+                getLogger().error("Failed to process data due to {} for input {}",
+                        new Object[]{exception.getLocalizedMessage(), input}, exception);
+                session.transfer(flowFile, REL_FAILURE);
+            context.yield();
```

Comment:
I don't think context.yield() is appropriate here. There is no condition that is occurring that would result in the processor unable to perform its task for some period of time, after which it would work again. That's what context.yield() is for. In this case, it is likely that the processor is misconfigured or that the data is invalid, correct? So in that case, I would just route to failure and continue on.

Reply:
Removed yield.  Thanks for the comments.